### PR TITLE
intersectSphere: Moved calculation of tca to after the first possible…

### PR DESCRIPTION
**Pull request by EliasHasle**
… early return

Just good practice to avoid a few (or a lot of) small, unnecessary dot product calculations.

Related issue: #XXXX

**Description**

A clear and concise description of what the problem was and how this pull request solves it.

*This contribution is funded by [Example](https://example.com)*

### CODE REVIEW

File: src/math/Ray.js [hunk 1]
```diff
@@ -220,12 +220,13 @@ class Ray {
 	intersectSphere( sphere, target ) {
 
 		_vector.subVectors( sphere.center, this.origin );
-		const tca = _vector.dot( this.direction );
+
 		const d2 = _vector.dot( _vector ) - tca * tca;
```
Comment (LeviPesin) [inline src/math/Ray.js:224]: `tca` is used on this line.

**Comment by Mugen87** [comment]
Please make sure in future PRs not to target `master`.

File: src/math/Ray.js (hunk 1 above)
Comment (EliasHasle) [reply src/math/Ray.js:224]: Indeed. I can't say I understand why, though. But judging from my experience today, it seems like a good idea to not challenge it right away... 😳🤔

**Comment by EliasHasle** [comment]
> Please make sure in future PRs not to target `master`.

OK. Sorry. (I think I forgot it once before. Next time, I should get a tatoo!)

**Pull request closed without merging**